internal/downloader/s3_test.go

```diff
@@ -0,0 +1,81 @@
+package downloader
+
+import (
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type mockS3 struct {
+	mock.Mock
+}
+
+func (m *mockS3) Download(w io.WriterAt, input *s3.GetObjectInput, options ...func(*s3manager.Downloader)) (n int64, err error) {
+	args := m.Called(w, input)
+	return int64(args.Int(0)), args.Error(1)
+}
+
+func (m *mockS3) DownloadWithContext(ctx aws.Context, w io.WriterAt, input *s3.GetObjectInput, options ...func(*s3manager.Downloader)) (n int64, err error) {
+	args := m.Called(ctx, w, input)
+	return int64(args.Int(0)), args.Error(1)
+}
+
+func (m *mockS3) DownloadWithIterator(ctx aws.Context, iter s3manager.BatchDownloadIterator, opts ...func(*s3manager.Downloader)) error {
+	args := m.Called(ctx, iter)
+	return args.Error(0)
+}
+
+func TestNewDownloader(t *testing.T) {
```

Comment:
These tests do not do very much besides assert that the `Download()` function behaves as expected. I am open to suggestions on what else could be tested.